common/config/flags.go

```diff
@@ -0,0 +1,175 @@
+package config
+
+import (
+	"github.com/urfave/cli"
+)
+
+// todo: create a PR to https://github.com/urfave/cli or create a helper function
+// to parse the canonical name of a flag, e.g. "p, port" is "port".
+
+// common flags
+var (
+	FlagDebug = cli.BoolFlag{
+		Name:   "debug",
+		EnvVar: "LAYER0_DEBUG",
+		Usage:  "show debug output",
+	}
+	FlagEndpoint = cli.StringFlag{
+		Name:   "endpoint",
+		Value:  DefaultEndpoint,
+		EnvVar: "LAYER0_ENDPOINT",
```

Comment:
Something I'm concerned about is the renaming of the environment variables themselves. This used to be called `LAYER0_API_ENDPOINT`. I'm concerned that by doing this, we are rendering older api image versions obsolete, because users would be forced to upgrade and change their environment variable structure. Other environment variables further below are also name changes.

Reply:
If we weren't in the process of a new minor version I would agree with you, but this refactor is already a minor version bump. Older versions of the API, CLI, and Setup binaries are already incompatible with this version in a number of different ways. 

API images are tightly coupled with their respective version of `l0-setup`. The upgrade/changes to their environment variable structure is done for them via the `l0-setup`, which handled with `l0-setup init` or `l0-setup upgrade`. 

So, within minor version updates I agree we should avoid changing these variables, but I see no problem with doing it between minor version updates.

Reply:
This is a good comment to think about if only because we're going to have more breaking changes in this release than we have had in a very long time. Historically, this has caused us headaches for people that don't upgrade across the board, so something to keep in mind when we update docs, etc.